base/math.jl

```diff
@@ -103,17 +103,19 @@ function evalpoly(x, p::Tuple)
         _evalpoly(x, p)
     end
 end
+evalpoly(x, ::Tuple{}) = zero(one(x)) # dimensionless zero, i.e. 0 * x^0
 
 evalpoly(x, p::AbstractVector) = _evalpoly(x, p)
 
 function _evalpoly(x, p)
     Base.require_one_based_indexing(p)
     N = length(p)
-    ex = p[end]
+    @inbounds p0 = N == 0 ? reduce_empty_iter(+, p) : p[N]
```

Comment:
Yes, there is a bounds check for `p[N]`  without `@inbouds` here.

In my mind, applying `@inbounds` to the entire line is *easier* to read.   And it doesn't affect any of the functions here except for `p[N]`.

Regarding `iszero`, `N = length(p)` should be an `Int`, so `iszero` has no particular advantage here.  (I'm also not sure `iszero` is actually faster for `BigInt`, by the way, since `BigInt(N) == 0` ends up calling a specialized method — it doesn't promote `0` to `BigInt`.